src/factory/useProfileLoader/visualLoader.js

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react';
```

Comment:
Since it's no longer tied to `useProfileLoader`, I think `visualLoader` should be in the `factory/utils` and it should be return `visualData` and not `chartData`.

Reply:
:+1: